ddl/db_partition_test.go

```diff
@@ -257,6 +266,42 @@ func (s *testIntegrationSuite) TestCreateTableWithHashPartition(c *C) {
 	partition by hash( year(hired) ) partitions 4;`)
 }
 
+func (s *testIntegrationSuite) TestCreateTableWithRangeColumnPartition(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test;")
+	tk.MustExec("drop table if exists log_message_1;")
+	tk.MustExec("set @@session.tidb_enable_table_partition = 1")
+	tk.MustExec(`
+create table log_message_1 (
+    add_time datetime not null default '2000-01-01 00:00:00',
+    log_level int unsigned not null default '0',
+    log_host varchar(32) not null,
+    service_name varchar(32) not null,
+    message varchar(2000)
+) partition by range columns(add_time)(
+    partition p201403 values less than ('2014-04-01'),
+    partition p201404 values less than ('2014-05-01'),
+    partition p201405 values less than ('2014-06-01'),
+    partition p201406 values less than ('2014-07-01'),
+    partition p201407 values less than ('2014-08-01'),
+    partition p201408 values less than ('2014-09-01'),
+    partition p201409 values less than ('2014-10-01'),
+    partition p201410 values less than ('2014-11-01')
+)`)
```

Comment:
Insert some records and do "select * from log_message_1"?

Reply:
1. This function just `TestCreateTableWithRangeColumnPartition`, not CRUD
2. Actually I manually test some DML statements
3. I should have port some mysql test for partition if this is not a special week and table partition have the lower priority. Porting mysql test will cover here.